### PR TITLE
Day7

### PR DESCRIPTION
**Pull request by amrendra-rathore**
### Task Description:

Define .env file with following properties:

PORT=9000
NODE_ENV=dev

Create a folder src

Create a file Server.ts inside src

Define a class with the following methods and properties:

- config: property, It will be set by constructor from parameter received.
- bootstrap: call different methods needed for bootstrapping of app.
It will call setupRoutes for now.
It will return this.
- setupRoutes: Method to setup routes
We will define a route(of type get) '/health-check' which will return 'I am OK' as response.
- run: Method to run the express app.
Log error in case of error
Log success message in case of success.
It will return this.

Export Server class.

Create a folder config inside src
- Create a file IConfig.ts which contains interface of config (properties like port and env)
- Create a file configuration.ts
Use dotenv module for extracting config from .env file.
Define config using interface(IConfig).
Use Object.freeze for protecting config object against any modifications after the declaration.
Export config.

Create a file index.ts
Import config and Server.
Create an instance of Server by passing config in constructor
Call methods bootstrap and run using instance
App should be running and success message.

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+PORT=9000
+NPDE_ENV=dev
```
Comment (divyansh0777) [inline .env]: Do not push .env file into GitHub. For reference create .env.example file with blank values and push it to GitHub.

File: package.json [hunk 2]
```diff
@@ -2,15 +2,16 @@
...
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
+    "serve": "nodemon src/index.ts",
     "build": "npx babel-node extra",
     "start:extra": "babel-node extra",
     "start:extraTs": "tsc-watch --project . --onSuccess \"npm run nodemon\"",
     "nodemon": "nodemon ./dist/index.js",
     "lint": "./node_modules/tslint/bin/tslint -c tslint.json -p tsconfig.json --exclude 'node_modules/**/*.{j,t}s'",
-    "lint:fix": "./node_modules/tslint/bin/tslint --fix -c tslint.json -p tsconfig.json --exclude 'node_modules/**/*.{j,t}s'" ,
+    "lint:fix": "./node_modules/tslint/bin/tslint --fix -c tslint.json -p tsconfig.json --exclude 'node_modules/**/*.{j,t}s'",
```
Comment (divyansh0777) [inline package.json]: linting command is not correct. Refer docs

File: package.json [hunk 3]
```diff
@@ -28,18 +29,22 @@
...
     "husky": "^8.0.1",
     "nodemon": "^1.3.3",
+    "ts-node": "^10.8.1",
     "tsc-watch": "^5.0.3",
     "typescript": "^4.7.4"
   },
   "dependencies": {
     "commitizen": "^1.0.4",
+    "dotenv": "^16.0.1",
+    "express": "^4.18.1",
```
Comment (divyansh0777) [inline package.json]: remove all carrot signs ^

File: src/Server.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  constructor(config){
+    this.config = config;
+    this.app = express();
+  }
+  public bootstrap(){
+    this.setupRoutes();
+    return this;
+  }
+  public setupRoutes()  {
+    this.app.get("/health-check", function(req , res)  {
```
Comment (divyansh0777) [inline src/Server.ts]: Use arrow functions instead of basic JS function syntax

File: src/config/IConfig.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+export interface IConfig  {
+  nodeEnv: any;
```
Comment (divyansh0777) [inline src/config/IConfig.ts]: Do not use any. Use string instead

File: src/config/configuration.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import { config } from "dotenv";
+import { IConfig } from "./IConfig";
+config();
+const configurations: IConfig = {
+  nodeEnv: process.env.NODE_ENV,
+  port: Number(process.env.PORT)
```
Comment (divyansh0777) [inline src/config/configuration.ts]: Use parseInt function, this will increase complexity bus Number is a class

File: src/index.ts [hunk 7]
```diff
@@ -0,0 +1,5 @@
+import configurations from './config/configuration';
+import Server from "./Server";
+
+let server = new Server(configurations);
```
Comment (divyansh0777) [inline src/index.ts]: Do not use let, use const instead

**Review by divyansh0777 — CHANGES_REQUESTED** [review]

File: src/Server.ts (hunk 4 above)
Comment (amrendra-rathore) [reply src/Server.ts]: used arrow function instead of basic javascript function.

File: src/config/IConfig.ts (hunk 5 above)
Comment (amrendra-rathore) [reply src/config/IConfig.ts]: Used string instead of any

File: src/index.ts (hunk 7 above)
Comment (amrendra-rathore) [reply src/index.ts]: Used Const instead of let.

File: package.json (hunk 3 above)
Comment (amrendra-rathore) [reply package.json]: Removed all carrot signs (^)

File: .env (hunk 1 above)
Comment (amrendra-rathore) [reply .env]: Removed .env file and created a sample file named .env.example

File: package.json (hunk 2 above)
Comment (amrendra-rathore) [reply package.json]: fixed commands/scripts for linting issues.

File: src/config/configuration.ts (hunk 6 above)
Comment (amrendra-rathore) [reply src/config/configuration.ts]: Used parseInt() instead of Number class.

**Pull request merged**